packages/compass-components/src/components/index-keys.tsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import Badge, { Variant } from '@leafygreen-ui/badge';
+import { css } from '@leafygreen-ui/emotion';
+import Icon from '@leafygreen-ui/icon';
+import { spacing } from '@leafygreen-ui/tokens';
+
+type IndexDirection =
+  | -1
+  | 1
+  | '2d'
+  | '2dsphere'
+  | 'text'
+  | 'geoHaystack'
+  | number;
```

Comment:
The index-model describes one indexes: https://github.com/mongodb-js/compass/tree/main/packages/index-model#index-field-types

But there is a bit different list in the create modal: https://github.com/mongodb-js/compass/blob/main/packages/compass-indexes/src/components/create-index-field/create-index-field.jsx#L17

@mcasimir which one is correct? to they can be different?

Reply:
As I was saying on slack, this is one of those situations where being more tolerant with types could be better. 

The driver type (which we copy/pasted here) is meant for the `createIndex` option and while the values returned by the server should be aligned, that may not be true with a more recent server, which could return new unexpected values. 

Restricting the set of possible values like this is more dangerous than anything cause we would rely on something that is not actually true.

I recommend changing this type to `IndexDirection = number | string` and handling possible corner cases appropriately: for example just render the index name without any icon if we get nonsensical values like `2` or an empty string.

Another comment that I have here is that this type could be named a bit better since this is not really only a direction, for example we could consider calling it `IndexTypeOrDirection`